lib/std/atomic/Atomic.zig

```diff
@@ -0,0 +1,522 @@
+                        2 => "w",
+                        4 => "l",
+                        8 => "q",
+                        else => @compileError("Invalid atomic type " ++ @typeName(T)),
+                    };
+
+                    const old_bit = asm volatile (instruction ++ suffix ++ " %[bit], %[ptr]"
+                        : [result] "={@ccc}" (-> u8) // LLVM doesn't support u1 flag register return values
+                        : [ptr] "*p" (&self.value),
+                          [bit] "X" (@as(T, bit))
+                        : "cc", "memory"
+                    );
+
+                    return @intCast(u1, old_bit);
+                }
+
+                const mask = @as(T, 1) << bit;
+                const value = switch (op) {
+                    .Set => self.fetchOr(mask, ordering),
+                    .Reset => self.fetchAnd(~mask, ordering),
+                    .Toggle => self.fetchXor(mask, ordering),
+                };
+
+                return @boolToInt(value & mask != 0);
+            }
+        });
+    };
+}
+
+fn atomicIntTypes() []const type {
+    comptime var bytes = 1;
+    comptime var types: []const type = &[_]type{};
+    inline while (bytes <= @sizeOf(usize)) : (bytes *= 2) {
+        types = types ++ &[_]type{std.meta.Int(.unsigned, bytes * 8)};
+    }
+    return types;
+}
+
+test "Atomic.loadUnchecked" {
+    inline for (atomicIntTypes()) |Int| {
+        var x = Atomic(Int).init(5);
+        try testing.expectEqual(x.loadUnchecked(), 5);
+    }
+}
+
+test "Atomic.storeUnchecked" {
+    inline for (atomicIntTypes()) |Int| {
+        var x = Atomic(usize).init(5);
+        x.storeUnchecked(10);
+        try testing.expectEqual(x.loadUnchecked(), 10);
+    }
+}
+
+test "Atomic.load" {
+    inline for (atomicIntTypes()) |Int| {
+        inline for (.{ .Unordered, .Monotonic, .Acquire, .SeqCst }) |ordering| {
+            var x = Atomic(Int).init(5);
+            try testing.expectEqual(x.load(ordering), 5);
+        }
+    }
+}
+
+test "Atomic.store" {
+    inline for (atomicIntTypes()) |Int| {
+        inline for (.{ .Unordered, .Monotonic, .Release, .SeqCst }) |ordering| {
+            var x = Atomic(usize).init(5);
+            x.store(10, ordering);
+            try testing.expectEqual(x.load(.SeqCst), 10);
+        }
+    }
+}
+
+const atomic_rmw_orderings = [_]Ordering{
+    .Monotonic,
+    .Acquire,
+    .Release,
+    .AcqRel,
+    .SeqCst,
+};
+
+test "Atomic.swap" {
+    inline for (atomic_rmw_orderings) |ordering| {
+        var x = Atomic(usize).init(5);
+        try testing.expectEqual(x.swap(10, ordering), 5);
+        try testing.expectEqual(x.load(.SeqCst), 10);
+
+        var y = Atomic(enum(usize) { a, b, c }).init(.c);
+        try testing.expectEqual(y.swap(.a, ordering), .c);
+        try testing.expectEqual(y.load(.SeqCst), .a);
+
+        var z = Atomic(f32).init(5.0);
+        try testing.expectEqual(z.swap(10.0, ordering), 5.0);
```

Comment:
Skip the `f32` tests for AAarch64 or the build will fail with `LLVM ERROR: Cannot select: intrinsic %llvm.aarch64.stxr`.

Reply:
Now that #8922 is merged you can remove the AArch64 check, the tests should pass on both that and ARM now.